### PR TITLE
Laboratory work #1, Kapitanova Daria - 21FPL1

### PR DESCRIPTION
**Pull request by kapitanovadaria**

### CODE REVIEW

**Review by artyomtugaryov — COMMENTED** [review]
Hello, @kapitanovadaria! As a reminder, the deadline for the first lab is September 30th and you have less than 2 weeks to complete it. Please hurry to be on time.

File: lab_1_keywords_tfidf/main.py [hunk 1]
```diff
@@ -48,7 +67,13 @@ def calculate_frequencies(tokens: list[str]) -> Optional[dict[str, int]]:
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if isinstance(tokens, list) and all(isinstance(token, str) for token in tokens):
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:70]: again

File: lab_1_keywords_tfidf/main.py [hunk 2]
```diff
@@ -99,7 +139,18 @@ def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optio
...
-    pass
+    if isinstance(term_freq, dict) and all(isinstance(token, str) for token in term_freq.keys()) and all(
+            isinstance(value, float) for value in term_freq.values()) and isinstance(idf, dict) and all(
+            isinstance(token, str) for token in idf.keys()) and all(isinstance(value, float) for value in idf.values()):
+        tfidf_dict = {}
+        for token in term_freq.keys():
+            if token in idf.keys():
+                tfidf_dict[token] = term_freq.get(token) * idf.get(token)
+            else:
+                tfidf_dict[token] = term_freq.get(token) * log(47 / (0 + 1))
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:150]: why do you do  0+ 1?

File: lab_1_keywords_tfidf/main.py [hunk 3]
```diff
@@ -118,7 +169,20 @@ def calculate_expected_frequency(
...
     """
-    pass
+    if isinstance(doc_freqs, dict) and all(isinstance(token, str) for token in doc_freqs.keys()) and all(
+            isinstance(value, int) for value in doc_freqs.values()) and isinstance(corpus_freqs, dict) and all(
+            isinstance(token, str) for token in corpus_freqs.keys()) and all(
+            isinstance(value, int) for value in corpus_freqs.values()):
+        exp_freq_dict = {}
+        for token, value in doc_freqs.items():
+            value_corpus = corpus_freqs.get(token)
+            all_doc = sum(doc_freqs.values()) - value
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:179]: You can calculate `sum(doc_freqs.values())` only once before `for`

File: lab_1_keywords_tfidf/main.py [hunk 4]
```diff
@@ -156,4 +230,17 @@ def extract_significant_words(chi_values: dict[str, float], alpha: float) -> Opt
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if isinstance(chi_values, dict) and all(isinstance(token, str) for token in chi_values.keys()) and all(
+            isinstance(value, float) for value in chi_values.values()) and isinstance(alpha, float):
+        CRITERION = {0.05: 3.842, 0.01: 6.635, 0.001: 10.828}
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:235]: Why is this CAPS?

File: seminars/practice_1_intro.py [hunk 5]
```diff
@@ -28,7 +28,7 @@
 # third_num = first_num / second_num
 
 # Fix the first exception:
-second_num = 6
+second_num = 5
```
Comment (artyomtugaryov) [inline seminars/practice_1_intro.py:31]: revert to 6

File: lab_1_keywords_tfidf/main.py (hunk 2 above)
Comment (kapitanovadaria) [reply lab_1_keywords_tfidf/main.py:150]: that's how it was in the formula! but I will replace it

File: lab_1_keywords_tfidf/main.py (hunk 4 above)
Comment (kapitanovadaria) [reply lab_1_keywords_tfidf/main.py:235]: that's how it was in the readme too! but I will replace it too

File: lab_1_keywords_tfidf/main.py [hunk 6]
```diff
@@ -99,7 +137,17 @@ def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optio
...
+    if not isinstance(term_freq, dict) and all(isinstance(token, str) for token in term_freq.keys()) and all(
+            isinstance(value, float) for value in term_freq.values()) and isinstance(idf, dict) and all(
+            isinstance(token, str) for token in idf.keys()) and all(isinstance(value, float) for value in idf.values()):
+        return None
+    tfidf_dict = {}
+    for token in term_freq.keys():
+        if token in idf.keys():
+            tfidf_dict[token] = term_freq.get(token) * idf.get(token)
+        else:
+            tfidf_dict[token] = term_freq.get(token) * log(47 / 1)
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/main.py:149]: why `/1` ?

File: lab_1_keywords_tfidf/start.py [hunk 7]
```diff
@@ -16,21 +19,51 @@
     with open(TARGET_TEXT_PATH, 'r', encoding='utf-8') as file:
         target_text = file.read()
 
+    clean_text = clean_and_tokenize(target_text)
+    print(clean_text)
```
Comment (artyomtugaryov) [inline lab_1_keywords_tfidf/start.py:23]: remove prints

File: requirements.txt [hunk 8]
```diff
@@ -1 +0,0 @@
-
```
Comment (artyomtugaryov) [inline requirements.txt:1]: remove the empty line

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**